### ISSUE TITLE
bi-directional variables don't work with persistency layer

### ISSUE THREAD

**Issue by killenb** [ChimeraTK/ControlSystemAdapter#20]
This problem shows when using the OPC-UA adapter with bi-directional variables, but it is probably because it is using the build-in persistency layer, in contrast to the doocs adapter.

Error message:
<pre>
terminate called after throwing an instance of 'ChimeraTK::logic_error'
  what():  This device side of a process array must not be associated with a persistent data storage.
</pre>

**Comment by killenb** [comment]
There seems to be a logical bug in the Bidirectional process array:

```
if(_allowPersistentDataStorage) {
  throw ChimeraTK::logic_error("This device side of a process array must not be associated with a "
                               "persistent data storage.");
}
```

It should be `if (!allowPersistentDataStorage) throw`. There is a test missing for this, which has to be added.

**Comment by killenb** [comment]
Fixed by 0626c70
